lib/bib.rb

```diff
@@ -1,10 +1,12 @@
 require 'httparty'
 
 class Bib < MarcRecord
+  @@mixed_bib_ids = nil
+
   def is_research?
     begin
       items = get_platform_api_data items_path
-      result = is_partner? || has_zero_items?(items) || has_at_least_one_research_item?(items)
+      result = is_partner? || first_item_is_research?(items) || is_mixed_bib?
```

Comment:
As discussed IRL, this looks right to me, and we should discuss ways to optimize for different scenarios. Ideas include:
 - Parameterize the "scrictness" of the is-research check so that we can explicitly skip the existence check for mixed bibs and bibs with 0 items (and for partner records, for which we're already skipping the existence check in all cases)
 - move the is-mixed-bib check first to prioritize a static lookup for those bibs (~0.6%) that we know are research without any db lookups